libs/wire-api/test/unit/Test/Wire/API/User.hs

```diff
@@ -21,18 +21,39 @@
 module Test.Wire.API.User where
 
 import Data.Aeson
+import qualified Data.Aeson as Aeson
 import Data.Aeson.Types as Aeson
+import Data.Domain
+import Data.Id
+import Data.Qualified
+import qualified Data.Text as Text
+import qualified Data.UUID.V4 as UUID
 import Imports
 import Test.Tasty
 import Test.Tasty.HUnit
-import Wire.API.User (parseIdentity)
-import Wire.API.User.Identity
+import Wire.API.User
 
 tests :: TestTree
 tests = testGroup "User (types vs. aeson)" $ unitTests
 
 unitTests :: [TestTree]
-unitTests =
+unitTests = parseIdentityTests ++ jsonNullTests
+
+jsonNullTests :: [TestTree]
+jsonNullTests = [testGroup "JSON null" [testCase "userProfile" $ testUserProfile]]
+
+testUserProfile :: Assertion
+testUserProfile = do
+  uid <- Id <$> UUID.nextRandom
+  let domain = Domain "example.com"
+  let colour = ColourId 0
+  let userProfile = UserProfile (Qualified uid domain) (Name "name") (Pict []) [] colour False Nothing Nothing Nothing Nothing Nothing Nothing
+  let profileJSONAsText = Text.pack $ show $ Aeson.encode userProfile
+  let msg = Text.unpack $ "toJSON encoding must not convert Nothing to null, but instead omit those json fields for backwards compatibility. UserProfileJSON:" <> profileJSONAsText
+  assertBool msg (not $ "null" `Text.isInfixOf` profileJSONAsText)
```

Comment:
I would have just pasted the offending json into a QQ-block, fixed it, and make a round-trip test from json to haskell and back out of it.  But this works, too.

Reply:
We already have JSON roundtrip tests. They don't catch this problem, so they are somewhat useless, as they don't make a difference between fields being omitted and fields being set to null, since Aeson accepts either and interprets them the same.
I'm not sure how you can write a roundtrip test to fail when there are null values. Do you have an example?

Reply:
Yes, I read your PR :)  i meant:

```
sample = [json|{"field": 3}|] -- omitting the null fields
test = encode (decode sample) `shouldBe` sample
```

Reply:
But I'm not saying you should change the test you have, which works fine.  Just thinking out loud.

Reply:
the more i think about it, the better i like my approach.  it's very easy to read, and it potentially catches other mistakes we haven't even thought about here.  on the other hand, it gives you more room for errors when crafting the samples.  donno.  but i want to have 2 or 3 samples for every type now, everywhere.  :)